framework/scripts/generate_test_list.py

```diff
@@ -86,11 +86,13 @@ def main():
             f.write(',\n'.join(test_list))
             f.write('\n};\n\n')
         # write selector function
-        f.write('optional<const span<struct test * const>> get_test_list(const char *test_list_name) {\n')
+        f.write('optional<const span<struct test * const>> get_test_list(const char *test_list_name, bool fallback_to_default) {\n')
         if default_name is not None:
             f.write(f'    if (!test_list_name) return {default_name}_test_list;\n')
```

Comment:
if no default list exists, this will cause SEG_FAULTS. Add appropriate if before compares.